docs/_docs/reference/contextual/derivation.md

```diff
@@ -26,30 +26,48 @@ given [T: Show]     : Show[Tree[T]]     = Show.derived
 
 We say that `Tree` is the _deriving type_ and that the `Eq`, `Ordering` and `Show` instances are _derived instances_.
 
-### Types supporting `derives` clauses
+More formally, when a class/trait/object/enum `DerivingType` `derives` a type class `TC`, the following given instance is created in `DerivingType`'s companion object (or `DerivingType` itself if it is an object):
+* if `DerivingType` doesn't have type parameters
+```scala
+given TC[DerivingType] = TC.derived
+```
+* if `DerivingType` has type parameters `[T_1, ... T_N]`
+```scala
+given [T_1: TC, ... T_N: TC]: TC[DerivingType[T_1, ... T_N]] = TC.derived
+```
+
+`TC.derived` is then in charge of creating an instance of `TC[DerivingType]`, potentially using term and/or type inference.
+
+**Note:** `TC.derived` is a normal method call, therefore if there are multiple definitions of `TC.derived`, overloading resolution applies.
+
+**Note:** `TC.derived` can be used manually, for example to implement `Ordering` for `Option`:
+
+```scala
+given [T: Ordering]: Ordering[Option[T]] = Ordering.derived
```

Comment:
please noo

Reply:
I realise this was moved around, but I suppose it should be strongly recommended to use `derives` clause as much as possible